app/services/invoices/update_service.rb

```diff
@@ -82,7 +82,7 @@ def handle_prepaid_credits(payment_status)
       return unless invoice.invoice_type&.to_sym == :credit
       return unless %i[succeeded failed].include?(payment_status.to_sym)
 
-      Invoices::PrepaidCreditJob.perform_later(invoice, payment_status)
+      Invoices::PrepaidCreditJob.perform_later(invoice, payment_status.to_sym)
```

Comment:
What about handling this with a `lock_key_arguments` defined in the job itself?

It would reduce the risk a forget about it :)

Reply:
Note: that it could be tackled later

Reply:
I will add it as well, thanks.

Reply:
It is interesting that job uniqueness mechanism does not consider the default value of the second argument and only consider the actual arguments passed in the perform call.